src/Core/Content/Media/Pathname/PathGenerator.php

```diff
@@ -0,0 +1,47 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Content\Media\Pathname;
+
+use Shopware\Core\Content\Media\Aggregate\MediaThumbnail\MediaThumbnailEntity;
+use Shopware\Core\Content\Media\MediaEntity;
+use Shopware\Core\Content\Media\Pathname\PathnameStrategy\PathnameStrategyInterface;
+use Shopware\Core\Framework\Plugin\Exception\DecorationPatternException;
+
+class PathGenerator extends AbstractPathGenerator
+{
+    private PathnameStrategyInterface $pathnameStrategy;
+
+    public function __construct(PathnameStrategyInterface $pathnameStrategy)
```

Comment:
Why do we need this abstraction over an abstraction?

Reply:
And this file is not covered by tests

Reply:
> Why do we need this abstraction over an abstraction?

Because of `Circular reference` :-)